### PR TITLE
Update GH Actions Test and Build Workflows

### PR DESCRIPTION
**Pull request by sestevens**
### Summary of changes

- Use actions/checkout@v2 instead of v1
- Use Xcode 12.4 instead of 12.3
- Update Carthage check to build the CarthageTest demo app. This is a more thorough check since it pulls down our dependencies (CardinalMobile and PPRiskMagnes).
- Remove `--allow-warnings` flag from `pod lib lint` so that it fails if there are warnings
- Allow Test and Build workflows to be run manually as well as on pull requests. This way we can run them manually prior to releasing.

### Checklist

- ~Added a changelog entry~

- @sestevens

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -1,38 +1,49 @@
...
-      - name: Run carthage build
-        run: carthage build --no-skip-current
+        run: |
+          git checkout $GITHUB_HEAD_REF
+          rm -rf SampleApps/SPMTest
+          git add SampleApps/SPMTest
+          git commit -m 'Remove SPMTest app to avoid Carthage timeout'
+      - name: Use current branch
+        run: |
+          echo 'git "file://'$PWD'" "'$GITHUB_HEAD_REF'"' > SampleApps/CarthageTest/Cartfile
```
Comment (scannillo) [inline .github/workflows/build.yml:34]: Does this need to be two lines?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -1,38 +1,49 @@
...
+      - name: Run carthage update
+        run: cd SampleApps/CarthageTest && carthage update
```
Comment (scannillo) [inline .github/workflows/build.yml:36]: This could be a separate PR - but maybe we can require our repo uses [Carthage 0.37.0](https://github.com/Carthage/Carthage/releases/tag/0.37.0), and we can get rid of that `carthage.sh` script (which is only used in the Rakefile at this point).

File: .github/workflows/build.yml (hunk 1 above)
Comment (sestevens) [reply .github/workflows/build.yml:34]: I can make it one

Comment (sestevens) [reply .github/workflows/build.yml:34]:
```suggestion
        run: echo 'git "file://'$PWD'" "'$GITHUB_HEAD_REF'"' > SampleApps/CarthageTest/Cartfile
```

**Review by scannillo — APPROVED** [review]

**Review by sshropshire — APPROVED** [review]
